src/main/java/com/ldtteam/blockout/controls/ImageRepeatable.java

```diff
@@ -0,0 +1,155 @@
+package com.ldtteam.blockout.controls;
+
+import com.ldtteam.blockout.Pane;
+import com.ldtteam.blockout.PaneParams;
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.Tuple;
+import org.lwjgl.opengl.GL11;
+
+/**
+ * Image element with repeatable middle part.
+ */
+public class ImageRepeatable extends Pane
+{
+    public static final int MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE = 256;
+
+    protected ResourceLocation resourceLocation;
+    protected int u = 0;
+    protected int v = 0;
+    protected int uWidth = 0;
+    protected int vHeight = 0;
+    protected int uRepeat = 0;
+    protected int vRepeat = 0;
+    protected int repeatWidth = 0;
+    protected int repeatHeight = 0;
+    protected int fileWidth = MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE;
+    protected int fileHeight = MINECRAFT_DEFAULT_TEXTURE_IMAGE_SIZE;
+
+    /**
+     * Default Constructor.
+     */
+    public ImageRepeatable()
+    {
+        super();
+    }
+
+    /**
+     * Constructor used by the xml loader.
+     *
+     * @param params PaneParams loaded from the xml.
+     */
+    public ImageRepeatable(final PaneParams params)
+    {
+        super(params);
+        final String source = params.getStringAttribute("source", null);
+        if (source != null)
+        {
+            resourceLocation = new ResourceLocation(source);
+            loadMapDimensions();
+        }
+
+        PaneParams.SizePair size = params.getSizePairAttribute("uv", null, null);
```

Comment:
We should probably stick with the "imageoffset" and "imagesize" terminology for consistency. Personally I'd like to use the terms "offset" and "crop" when referring to the texture throughout all of blockout, but that's just my personal sense of clarity.

Reply:
I would personally rename the old params, since imageOffset is more like pane offset for me
crop means something different? or I haven't seen it being used in this context

Reply:
I have nothing wrong with offset personally. Could just name everything with a source prefix, i.e. ``sourcepos`` ``sourcesize``, which is the most boring but most clear for newcomers I think. Technically we are cropping out a section of the source, but you're right in that crop doesn't usually refer directly to the width and height. I have seen it used here before though, if only once. We could potentially code in a 4-hand property, e.g. ``crop="2 2 192 20"``, which is a more logical use of the word (but of course it's ambiguous whether it is point-to-point or pos and size).

I don't really mind which but I do think we should be consistent across the board.

Reply:
> We could potentially code in a 4-hand property, e.g. crop="2 2 192 20", which is a more logical use of the word (but of course it's ambiguous whether it is point-to-point or pos and size).

I have thought about this too, not sure if it's a good approach tho

We can also go back to traditional `textureoffset` and `texturesize`, and thus `repeatoffset `and `repeatsize`

Reply:
Yeah that's nice and clear, I like it.